pylxd/models/storage_pool.py

```diff
@@ -669,6 +669,36 @@ def api(self):
         """
         return self.volume.api[self._endpoint][self.name]
 
+    @classmethod
+    def __parse_snaspshot_json(cls, volume, snapshot_json):
+        snapshot_object = cls(volume.client, volume=volume, **snapshot_json)
+
+        # Snapshot names are namespaced in LXD, as volume-name/snapshot-name.
+        # We hide that implementation detail.
+        snapshot_object.name = snapshot_object.name.split("/")[-1]
+
+        # If reponse does not include expires_at sync the object to get that attribute.
+        if not snapshot_json.get("expires_at"):
+            snapshot_object.sync()
+
+        # Getting '0001-01-01T00:00:00Z' means that the volume does not have an expiration time set.
+        if snapshot_object.expires_at == "0001-01-01T00:00:00Z":
+            snapshot_object.expires_at = None
+
+        # Overriding default value for when created_at is not present on the response due to using LXD 4.0.
+        # Also having created_at as "0001-01-01T00:00:00Z" means this information is not available.
+        # This could be because the information was lost or the snapshot was created using an older LXD version.
+        if (
+            not volume.client.has_api_extension("storage_volumes_created_at")
+            or snapshot_json["created_at"] == "0001-01-01T00:00:00Z"
+        ):
+            snapshot_object.created_at = None
+
+        # This field is may empty so derive it from its volume.
+        snapshot_object.content_type = volume.content_type
```

Comment:
```suggestion
        if not snapshot_object.content_type:
            snapshot_object.content_type = volume.content_type
```

It feels safer than blindly overwriting the value, WDYT?

Reply:
I thought the same, ended up opting for this because this field should (if it doesn't it would propably be a bug) always have the same value as the volume so makes little difference. But either way is fine for me, I will push the changes.